python/tk_desktop/project_commands_model.py

```diff
@@ -212,8 +232,14 @@ def child_cmp(left, right):
             if util.is_version_older(left_version, right_version):
                 return 1
             return 0
-        children.sort(cmp=child_cmp)
-        return children
+
+        # No need to sort lists if they are empty.
+        if other_children:
```

Comment:
this feels like premature optimization to me. it's even possible that it's actually faster without the check :) I suggest removing it.